docs/pbc/all/dynamic-multistore/202404.0/tutorials-and-howtos/how-to-add-new-custom-field.md

```diff
@@ -0,0 +1,329 @@
+use Pyz\Zed\StoreContextGui\Communication\Form\StoreContextCollectionForm;
+use Symfony\Component\Form\FormBuilderInterface;
+
+class StoreContextFormExpander extends SprykerStoreContextFormExpander
+{
+    /**
+     * @param \Symfony\Component\Form\FormBuilderInterface $builder
+     * @param \Generated\Shared\Transfer\StoreTransfer $storeTransfer
+     *
+     * @return \Symfony\Component\Form\FormBuilderInterface
+     */
+    public function expand(FormBuilderInterface $builder, StoreTransfer $storeTransfer): FormBuilderInterface
+    {
+        $builder->add(
+            static::FIELD_APPLICATION_CONTEXT_COLLECTION,
+            StoreContextCollectionForm::class,
+            $this->getOptions(),
+        );
+
+        return $builder;
+    }
+}
+```
+
+*src/Pyz/Zed/StoreContextGui/Communication/StoreContextGuiCommunicationFactory.php*
+
+```php
+
+namespace Pyz\Zed\StoreContextGui\Communication;
+
+use Pyz\Zed\StoreContextGui\Communication\Expander\StoreContextFormExpander;
+use Spryker\Zed\StoreContextGui\Communication\Expander\StoreContextFormExpanderInterface;
+use Spryker\Zed\StoreContextGui\Communication\StoreContextGuiCommunicationFactory as SprykerStoreContextGuiCommunicationFactory;
+
+class StoreContextGuiCommunicationFactory extends SprykerStoreContextGuiCommunicationFactory
+{
+    /**
+     * @return \Spryker\Zed\StoreContextGui\Communication\Expander\StoreContextFormExpanderInterface
+     */
+    public function createStoreContextFormExpander(): StoreContextFormExpanderInterface
+    {
+        return new StoreContextFormExpander(
+            $this->createStoreContextFormDataProvider(),
+        );
+    }
+}
+```
+
+## Setup and configure the store settings in the Back Office
+
+1. In the Back Office, go to **Administration**.
+2. On the **Stores** page choice store and click **Edit store**.
+3. In the **Edit store** page, go to the **Settings** tab.
+4. In the **Settings** tab, find the new field **Contact Email**.
+
+![configure-application-timezone](/images/dynamic-multistore/screen2.png)
+
+Each store must be have one default application block.
+
+5. Click **Save** to save the changes.
+
+
+## How extract new field value from StoreTransfer
+
+This is simple example how to extract the new field `contactEmail` value from `StoreTransfer` in the project.
+
+```php
+
+// .... 
+
+function extractContactEmail(StoreTransfer $storeTransfer): ?string
+{
+
+    foreach ($storeTransfer->getApplicationContextCollectionOrFail()->getApplicationContexts() as $storeApplicationContextTransfer) {
+        if ($storeApplicationContextTransfer->getApplication() === APPLICATION && $storeApplicationContextTransfer->getContactEmail() !== null) {
+            return $storeApplicationContextTransfer->getContactEmail();
+        }
+
+        if ($storeApplicationContextTransfer->getApplication() === null) {
+           return $storeApplicationContextTransfer->getContactEmail();
+      }
+    }
+}
+
+// ....
```

Comment:
Please do it similarly as done with timezone, as part of `Store` transfer + expander plugin that does this.

Reply:
Added example for use plugin extender for contact email.